storage/common/include/algebra/storage/vector.hpp

```diff
@@ -0,0 +1,246 @@
+/** Algebra plugins, part of the ACTS project
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// Project include(s)
+#include "algebra/qualifiers.hpp"
+#include "algebra/storage/array_operators.hpp"
+
+// System include(s).
+#include <cstddef>
+#include <initializer_list>
+#include <type_traits>
+#include <utility>
+
+namespace algebra::storage {
+
+/// Vector wrapper for AoS vs interleaved SoA data. @c value_t can e.g. be a
+/// SIMD vector.
+template <std::size_t N, typename value_t,
+          template <typename, std::size_t> class array_t>
+class vector {
+ public:
+  // Value type is a simd vector in SoA and a scalar in AoS
+  using value_type = value_t;
+  /// Underlying data array type
+  using array_type = array_t<value_t, N>;
+
+  /// Default contructor sets all entries to zero.
+  constexpr vector() { zero_fill(std::make_index_sequence<N>{}); }
+
+  /// Construct from element values @param vals .
+  ///
+  /// In order to avoid uninitialized values, which deteriorate the performance
+  /// in explicitely vectorized code, the underlying data array is filled with
+  /// zeroes if too few arguments are given.
+  template <typename... Values,
+            std::enable_if_t<
+                std::conjunction_v<std::is_convertible<Values, value_type>...>,
+                bool> = true>
+  constexpr vector(Values &&... vals) : m_data{std::forward<Values>(vals)...} {
```

Comment:
I'm fine with this type of SFINAE, this part is good on my part. In the morning I was also trying to make a similar thing work, but couldn't quite find the right incantation. So kudos to @stephenswat.

But there was one thing that Copilot very much wanted me to do while playing with this code: A check for `sizeof...(Values) <= N`. I think you could just add it with an `$$` after the `std::conjunction_v` expression.

Reply:
Done